SETUP.md

```diff
@@ -71,69 +71,71 @@ Finally, flick the repository's switch to enable CI.
-  -N "" \
+  -t rsa -b 4096 -N "" \
+  -C "deploy@travis-ci.com" \
   -f deploy.key
+```
+
+#### Add the public key to GitHub
 
-# For convenience, print the URL to add the public key to GitHub
-echo https://github.com/$OWNER/$REPO/settings/keys
+```sh
+# Print the URL for adding the public key to GitHub
+echo "https://github.com/$OWNER/$REPO/settings/keys/new"
+
+# Print the public key for copy-pasting to GitHub
+cat deploy.key.pub
 ```
 
-Manually add the text of `deploy.key.pub` (with write access) to GitHub under the repository's deploy key settings (the URL echoed above).
-Give the key a descriptive title, such as "Travis CI Manubot".
+Go to the GitHub settings URL echoed above in a browser, and click "Add deploy key".
+For "Title", add a description like "Manubot Travis Deploy Key".
+Copy-paste the contents of the `deploy.key.pub` text file (printed above) into the "Key" text box.
+Check the "Allow write access" box below.
+Finally, click "Add key".
 
-For the next step, you need the [Travis command line client](https://github.com/travis-ci/travis.rb) installed.
-This program is a Ruby gem:
-install it with `gem install travis` (not `apt install travis`, which is a different program).
-After the install, you will need to provide your credentials to login to travis with `travis login`.
+#### Add the private key to Travis CI
 
 ```sh
-travis encrypt-file \
-  --com \
-  --repo=$OWNER/$REPO \
-  --force \
-  deploy.key > travis-encrypt-file.log
+# Print the URL for adding the private key to Travis CI
+echo "https://travis-ci.com/$OWNER/$REPO/settings"
 ```
 
-`ci/travis-encrypt-file.log` will contain a line such as:
+Next, go to the Travis CI repository settings page (URL echoed above).
+Add a new record in the "Environment Variables" section.
+For "NAME", enter `MANUBOT_SSH_PRIVATE_KEY`.
+Next we will encode the text in `deploy.key` to remove newlines, which is required for entry into the Travis settings.
+Run any of the following commands (whichever works), and copy-paste the output into "VALUE":
 
-```
-openssl aes-256-cbc -K $encrypted_8cc93e35fb85_key -iv $encrypted_8cc93e35fb85_iv -in test.key.enc -out test.key -d
-```
+```shell
+# For systems with Python (2 or 3) installed
+python -c "import base64; print(base64.standard_b64encode(open('deploy.key', 'rb').read()).decode())"
 
-`ci/deploy.sh` must be updated with the random string from this line (i.e. `8cc93e35fb85`).
-The following command automates this substitution.
+# For Windows or Linux systems that have the GNU coreutils base64 command
+base64 --wrap=1000000 deploy.key
 
-```sh
-# Edit ci/deploy.sh with travis secure env variables generated by travis encrypt-file
-TRAVIS_ENCRYPT_ID=`python -c "import re; \
-  text = open('travis-encrypt-file.log').read(); \
-  print(re.search('encrypted_([a-f0-9]+)_key', text).group(1))"`
-sed "s/9befd6eddffe/$TRAVIS_ENCRYPT_ID/g" deploy.sh > tmp && mv -f tmp deploy.sh
+# For macOS systems
+base64 --break=1000000 deploy.key
```

Comment:
Another command that I can think of is to use `base64` subcommand bundled in openssl:
```
openssl base64 -A < deploy.key  # `-A` generates base64 output as a single line
```

This command is probably identical on Linux/Windows/macOS (I didn't test openssl that is bundled in git for Windows though).

Reply:
Interesting idea to use `openssl base64`, although it [seems that](https://superuser.com/a/769829/854334) the options on Linux and Mac may be somewhat different.

Reply:
Hmm, I wonder whether `-A` is supported by openssl on macOS.

Reply:
Does the presence of `ssh-keygen` on a system guarantee that `openssl` is [installed](https://serverfault.com/a/780503/519601)? If so, perhaps we should just use `openssl` to create `deploy.key.base64` and have the user copy the text of that?

Reply:
My version of Git Bash for Windows does have `openssl` so I could help test that.

Reply:
We've now switched to `openssl base64`. Including the alternative implementations below in case we need them in the future:

```shell
# For systems with Python (2 or 3) installed
python -c "import base64; print(base64.standard_b64encode(open('deploy.key', 'rb').read()).decode())"

# For Windows or Linux systems that have the GNU coreutils base64 command
base64 --wrap=1000000 deploy.key

# For macOS systems
base64 --break=1000000 deploy.key

# For systes with openssl
openssl base64 -A -in=deploy.key
```